x-pack/plugins/cloud_security_posture/public/components/fleet_extensions/integration_settings.tsx

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import React from 'react';
+import { EuiText, EuiFieldText, EuiFormRow, EuiSpacer } from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n-react';
+import type { PosturePolicyTemplate } from '../../../common/types';
+
+interface IntegrationSettingsInfoProps {
+  type: PosturePolicyTemplate;
+  showStepTitle: boolean;
+}
+
+export const IntegrationSettingsInfo = ({ type, showStepTitle }: IntegrationSettingsInfoProps) => (
+  <div>
+    {/* We need to add the step title in integration edit screen */}
+    {showStepTitle && (
+      <>
+        <EuiSpacer />
+        <EuiText>
+          <h4>
+            <FormattedMessage
+              id="xpack.csp.fleetIntegration.integrationSettingsTitle"
+              defaultMessage="Integration Settings"
+            />
+          </h4>
+        </EuiText>
+        <EuiSpacer />
+      </>
+    )}
+    <EuiText color={'subdued'} size="s">
+      {type === 'kspm' && (
+        <FormattedMessage
+          id="xpack.csp.fleetIntegration.configureKspmIntegrationDescription"
+          defaultMessage="Select the Kuberentes cluster type you want to monitor and then fill in the name and description to help identify this integration"
+        />
+      )}
+      {type === 'cspm' && (
+        <FormattedMessage
+          id="xpack.csp.fleetIntegration.configureCspmIntegrationDescription"
+          defaultMessage="Select the cloud service provider (CSP) you want to monitor and then fill in the name and description to help identify this integration"
+        />
+      )}
+    </EuiText>
+    <EuiSpacer />
+  </div>
+);
+
+interface IntegrationInfoFieldsProps {
+  name: string;
+  description: string;
+  onChange(field: string, value: string): void;
+}
+
+export const IntegrationSettings = ({
+  name,
+  description,
+  onChange,
+}: IntegrationInfoFieldsProps) => (
+  <div>
+    <EuiSpacer />
+    <Field
+      value={name}
+      id="name"
+      onChange={(value) => onChange('name', value)}
```

Comment:
can this function be memoized with `useCallback()`?

```js
const onChangeName = useCallback((value: string) => {
  onChange('name', value);
}, []);

<Field
	//...
	onChange={onChangeName}
/>
```

Reply:
(nit) This is probably premature optimization. The `useCallback` implementation here likely uses more memory, actually, since we have to 

1. Invoke `useCallback`
2. Define the callback function
3. Instantiate an array

vs the non `useCallback` implementation in which we
1. Define the callback function

This might be naive, since I don't know the internals of `onChange` as it's implemented by the EUI form field components, but I assume this is a simple key/value store in React state.

For single-line change handlers like this with no logic, `useCallback` adds readability overhead without tangible performance benefit. If instead of a single `onChange` we were doing some complex sorting logic, data marshaling, etc, I think `useCallback` would clearly be the right way to go. In this instance, though, I'll butt in and suggest avoiding the premature memoization.

It's not directly inside the Fleet plugin though so feel free to ignore me!